tool/tsh/common/tsh.go

```diff
@@ -2594,7 +2591,7 @@ func getNodeRow(proxy, cluster string, node types.Server, verbose bool) []string
 	return row
 }
 
-func printNodesAsText(output io.Writer, nodes []types.Server, verbose bool) error {
+func printNodesAsText[T types.Server](output io.Writer, nodes []T, verbose bool) error {
```

Comment:
IMO this does nothing except for make the code harder to read. Are generics necessary here when we're already generic over implementations of `types.Server`?

https://twitter.com/MattJamesBoyle/status/1690981292110741504

Reply:
Per the linked guidelines, two of the best reasons to employ generics are:

- When working with collections such as slices/maps.
- When you find yourself needing to reuse previously non-generic logic for a different type.

This function previously only worked with `[]types.Server`, but I needed to call it on `[]*types.ServerV2`.  Seems like the quintessential generics use case to me.